src/store/participantsStore.js

```diff
@@ -809,8 +833,23 @@ const actions = {
 		return false
 	},
 
-	async joinCall({ commit, getters }, { token, participantIdentifier, flags, silent, recordingConsent }) {
-		if (!participantIdentifier?.sessionId) {
+	async joinCall({ commit, getters, state }, { token, participantIdentifier, flags, silent, recordingConsent }) {
+		// SUMMARY: join call process
+		// There are 2 main steps to join a call:
+		// 1. Join the call (signaling-join-call)
+		// 2A. Wait for the users list (signaling-users-in-room) INTERNAL server event
+		// 2B. Wait for the users list (signaling-users-changed) EXTERNAL server event
+		// In case of failure, we receive a signaling-join-call-failed event
+
+		// Exception 1: We may receive the users list before the signaling-join-call event
+		// In this case, we use the isParticipantsListReceived flag to handle this case
+
+		// Exception 2: We may receive the users list in a second event of signaling-users-changed or signaling-users-in-room
+		// In this case, we always check if the list is the updated one (it has the current participant in the call)
+
+		const { sessionId } = participantIdentifier
```

Comment:
Just in case so we don't stuck here
```suggestion
		const { sessionId } = participantIdentifier ?? {}
```